src/repr/src/row/encoding.rs

```diff
@@ -424,4 +619,70 @@ mod tests {
         row.encode(&mut encoded);
         assert_eq!(Row::decode(&encoded), Ok(row));
     }
+
+    fn schema_and_row() -> (RelationDesc, Row) {
+        let row = Row::pack(vec![
+            Datum::True,
+            Datum::False,
+            Datum::True,
+            Datum::False,
+            Datum::Null,
+        ]);
+        let schema = RelationDesc::from_names_and_types(vec![
+            (
+                "a",
+                ColumnType {
+                    nullable: false,
+                    scalar_type: ScalarType::Bool,
+                },
+            ),
+            (
+                "b",
+                ColumnType {
+                    nullable: false,
+                    scalar_type: ScalarType::Bool,
+                },
+            ),
+            (
+                "c",
+                ColumnType {
+                    nullable: true,
+                    scalar_type: ScalarType::Bool,
+                },
+            ),
+            (
+                "d",
+                ColumnType {
+                    nullable: true,
+                    scalar_type: ScalarType::Bool,
+                },
+            ),
+            (
+                "e",
+                ColumnType {
+                    nullable: true,
+                    scalar_type: ScalarType::Bool,
+                },
+            ),
+        ]);
+        (schema, row)
+    }
+
+    #[test]
+    fn columnar_roundtrip() {
+        let (schema, row) = schema_and_row();
```

Comment:
don't think we need to go crazy with tests right away, but we should probably have tests that roundtrip more than 1 row

Reply:
note: Paul retracted this request offline in the interest of getting this merged now